### PR TITLE
Run test

### LINKED ISSUE #100: meson: enable tests
I don't think it should be that hard to get working in theory?

### PR DESCRIPTION
**Pull request by inkydragon**
Depend on #119, merge it first.

fix #100

### CODE REVIEW

**Comment by CoffeeFlux** [comment]
At a glance this looks fine, though it would be nice to get the tests running on CI as well. Will review more carefully once some of the other stuff gets merged and I can actually try this out locally.

File: tests/unset.bat [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cd %1
```
Comment (CoffeeFlux) [inline tests/unset.bat:1]: What is this script for? I don't see anything calling it?

File: tests/meson.build [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+test_inc = include_directories('support')
+
+# force UTF8 for MSVC
+if host_machine.system() == 'windows'
+    extra_args = ['/utf-8']
+else
+    extra_args = []
+endif
+
+e = executable(
```
Comment (CoffeeFlux) [inline tests/meson.build:64]:
```suggestion
runner = executable(
```

Not super picky about the exact name, but something better than `e` would be good.

File: tests/meson.build [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    'gtest-main',
+    all_test_sources,
+    include_directories : [test_inc, libaegisub_inc, deps_inc],
+    dependencies : all_test_deps,
+    cpp_args : extra_args,
+    link_with : all_test_dep_libs,
+)    
+test('gtest main', e)
```
Comment (CoffeeFlux) [inline tests/meson.build:72]:
```suggestion
test('gtest main', runner)
```

File: tests/meson.build [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+# setup test env
+if host_machine.system() == 'windows'
+    setup_sh = find_program('setup.bat')
+else
+    setup_sh = find_program('setup.sh')
+endif
+
+test_data_des = meson.project_build_root()
+custom_target('setup-test-data',
+    input: e,
```
Comment (CoffeeFlux) [inline tests/meson.build:84]:
```suggestion
    input: runner,
```

**Review by CoffeeFlux — APPROVED** [review]

File: tests/unset.bat (hunk 1 above)
Comment (inkydragon) [reply tests/unset.bat:8]: The `setup.bat` script in the same folder creates some read-only, no-access files.

This `unset.bat` script can help you restore permissions and delete the test data folder, when you want to rebuild (`meson build -C build --wipe`) or delete the entire build folder.

This script is not called anywhere, it is just a development tool

**Pull request merged**